src/reporter.ts

```diff
@@ -76,7 +76,19 @@ export const sendSpans = async (spans: any[], addEnrichmentSpan: boolean = true)
 };
 
 const isJsonContent = (payload: any, headers: Object) => {
-  return isString(payload) && headers['content-type'] && headers['content-type'].includes('json');
+  const isJsonString = (str: any) => {
+    try {
+      JSON.parse(str);
+      return true;
+    } catch (e) {
+      return false;
+    }
+  };
+
+  return (
+    (isString(payload) && headers['content-type'] && headers['content-type'].includes('json')) ||
+    isJsonString(payload)
```

Comment:
I suggest several improvements:
1. shorter conditional on the header
2. lowercase check, for cases it's not normalized
3. moving the string check to be first, so if it fails there no need to call `isJsonString` for no reason
```suggestion
    isString(payload) && (headers['content-type']?.toLowerCase().includes('json') || isJsonString(payload)
```